### PR TITLE
[Bot] Implement from issue #106: Create simple math utility with basic operations

### LINKED ISSUE #106: Create simple math utility with basic operations
﻿## Project Tag
project: math

## Description
Create a lightweight math utility module that provides basic arithmetic operations for use in other projects.

## Acceptance Criteria
- Function to add two numbers
- Function to multiply two numbers  
- Basic input validation
- Simple test to verify functions work
- Documentation with usage example

## Files to create
- \`math_utils.py\` - main utility functions
- \`test_math_utils.py\` - basic tests
- \`README.md\` - usage documentation

## Dependencies
- No external dependencies
- Use Python standard library only

## Scope & Segmentation Guidelines
**IMPORTANT FOR ANALYZER**: This is a simple, focused project. 
- Create maximum 3-4 logical tasks
- Group related activities together (don't create separate tasks for each function)
- Suggested segmentation: (1) Core implementation, (2) Testing, (3) Documentation
- Avoid micro-tasks like 'create __init__.py' or 'add single function'
- Keep tasks substantial and meaningful

This is a workflow verification test - prioritize logical task grouping over granular breakdown."

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #106

Project root: `projects`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-13 22:32:15 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of the math utility is functional but lacks handling for None inputs and could benefit from additional operations and improved test coverage.

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math_utils.py:9`**: The functions do not handle cases where inputs are None.
  *Suggestion*: Add a check for None inputs and raise a ValueError.

#### 💡 SUGGESTION
- **`projects/math_utils.py`**: Consider adding more mathematical operations for utility.
  *Suggestion*: Implement additional functions like subtract and divide.
- **`projects/test_math_utils.py`**: Test coverage could be improved.
  *Suggestion*: Add tests for edge cases, such as adding or multiplying by zero.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math_utils.py
+++ b/projects/math_utils.py
@@ -5,6 +5,10 @@
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
     return a + b
+
+    if a is None or b is None:
+        raise ValueError("Arguments cannot be None.")
+
 def multiply(a, b):
     """Return the product of two numbers."""
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
     return a * b
+
+    if a is None or b is None:
+        raise ValueError("Arguments cannot be None.")
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!

**Comment by morotrade** [comment]
❌ LLM diff validation failed in PR-fix mode:

```
Expected exactly 1 diff block, found 2. LLM must return a single unified diff.
```
